### PR TITLE
WIP: Cleanup amap TcGlobals

### PR DESCRIPTION
**Pull request by KevinRansom**
When reviewing a PR recently we noticed that I had added plumbing for a TcGlobals to an api that also has  (amap: Import.ImportMap) , which carries g a TcGlobals.

A quick perusal of the code showed that this was fairly common.  I just spent a while going through the code base and this is what I got.

There is still a ton more to do:

@dsyme … is this worthwhile?  should I continue?

### CODE REVIEW

File: src/fsharp/AccessibilityLogic.fs [hunk 1]
```diff
@@ -193,21 +193,21 @@ let CheckTyconReprAccessible amap m ad tcref =
      res)
             
 /// Indicates if a type is accessible (both definition and instantiation)
-let rec IsTypeAccessible g amap m ad ty = 
-    match tryAppTy g ty with
+let rec IsTypeAccessible (amap:Import.ImportMap) m ad ty =
```
Comment (forki) [inline src/fsharp/AccessibilityLogic.fs]: (amap: Import.ImportMap) <- needs space

Comment (KevinRansom) [reply src/fsharp/AccessibilityLogic.fs]: thx.

**Comment by dsyme** [comment]
> @dsyme … is this worthwhile? should I continue?

I've tended to not do this.  But I don't really mind if you do.

I've sometimes looked at combining TcGlobals and amap, but never got around to it, partly because TcImports uses this "base TcImports" and "derived TcImports" to help share resources.

**Comment by dsyme** [comment]
See https://github.com/dotnet/fsharp/issues/7212.  I think perhaps we should try this then change `g amap` --> `g` everywhere

**Comment by dsyme** [comment]
Yeah, looking at change like

    let rec IsTypeAccessible (amap:Import.ImportMap) m ad ty = 

I really don't like this.  We should just be aiming to get rid of amap.  I don't more and more `amap` scattered through the codebase.

**Comment by KevinRansom** [comment]
@dsyme,  my preference would be for tcGlobals to go everywhere, but there has been a tendency to hang them on context classes like amap, and cenv and nenv and pass those around.

The reason I am okay with making this change is because I am not adding amap anywhere, merely eliminating the tc globals, when they are doubled up.

Kevin

**Comment by KevinRansom** [comment]
@dsyme, closing this in favour of your PRto make TcGlobals per project.  Then we can take a pass at cleaning up apis, taking both.

Kevin

**Pull request closed without merging**